lib/src/dto/payload.dart

```diff
@@ -110,38 +115,53 @@ class Payload {
       "redirect_url": redirectUrl,
     };
 
-    putIfNotNull(map: json, key: "payment_type", value: paymentType);
-    putIfNotNull(map: json, key: "expirymonth", value: expiryMonth);
-    putIfNotNull(map: json, key: "expiryyear", value: expiryYear);
-    putIfNotNull(map: json, key: "cvv", value: cvv);
-    putIfNotNull(map: json, key: "cardno", value: cardNo);
-    putIfNotNull(map: json, key: "accountbank", value: bank?.code);
-    putIfNotNull(map: json, key: "bvn", value: bvn);
-    putIfNotNull(map: json, key: "accountnumber", value: accountNumber);
-    putIfNotNull(map: json, key: "passcode", value: passCode);
-    putIfNotNull(map: json, key: "phonenumber", value: phoneNumber);
-    putIfNotNull(map: json, key: "payment_plan", value: paymentPlan);
-    putIfNotNull(map: json, key: "billingzip", value: billingZip);
-    putIfNotNull(map: json, key: "pin", value: pin);
-    putIfNotNull(map: json, key: "suggested_auth", value: suggestedAuth);
-    putIfNotNull(map: json, key: "billingcity", value: billingCity);
-    putIfNotNull(map: json, key: "billingaddress", value: billingAddress);
-    putIfNotNull(map: json, key: "billingstate", value: billingState);
-    putIfNotNull(map: json, key: "billingcountry", value: billingCountry);
-    putIfNotNull(map: json, key: "billingzip", value: billingZip);
-    putIfTrue(map: json, key: "is_us_bank_charge", value: isUsBankCharge);
-    putIfTrue(
-        map: json, key: "is_mobile_money_franco", value: isMobileMoneyFranco);
-    putIfTrue(map: json, key: "is_mpesa", value: isMpesa);
-    putIfTrue(map: json, key: "is_mpesa_lipa", value: isMpesaLipa);
+    if (paymentMethod != null) {
```

Comment:
Please, can you remove this and the switch statement?

For `country` and` currency` fields, check if `init.acceptUgMobileMoneyPayments` in `ValidatorUtils#validateInitializer` is true and validate that both fields contain the right values for Uganda mobile money. If it doesn't meet this requirement, return an error message so it developer can set the right values. Please check [this example](https://github.com/wilburt/rave_flutter/blob/74ad1008d3f651ba87d546dcda4d8cea71e851f2/lib/src/common/validator_utills.dart#L160).

For  `is_mobile_money_ug` and `payment_type`, check [this line](https://github.com/wilburt/rave_flutter/blob/74ad1008d3f651ba87d546dcda4d8cea71e851f2/lib/src/dto/payload.dart#L136) for an example of how to pass it.